### ISSUE TITLE
Suggestion: Duplicate fields level and parentTag? in FireTag

### ISSUE THREAD

**Issue by abcdefguan** [cornell-dti/office-hours#524]
As of the current implementation, the fields level and parentTag? in FireTag are duplicates.

When parentTag is undefined, level is 1. When parentTag is undefined, level is 2.
This means that unless we're planning to support multilevel tags (ie level > 2), we don't need the level field in FireTag and can instead derive the level by looking at whether parentTag is defined.
If we eliminate the level field, we can avoid internal invariants on our data types and reduce the risk of breaking these internal invariants, which can cause bugs.

**Comment by ngwattcos** [comment]
Good point. I resolved a bug with these just last semester. I think it's better to restructure the data going forward with some kind of parenting or child relationships, not considering levels.
